### PR TITLE
[hotfix] Autoscaler's K8 support

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?

At some point our command parsing broke for kubernetes. We probably need some better from of integration testing to prevent this from happening in the future...
 
## Related issue number

#10759

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/command_runner.py [hunk 1]
```diff
@@ -220,6 +220,9 @@ def run(
                 cmd = _with_environment_variables(cmd, environment_variables)
             cmd = _with_interactive(cmd)
             final_cmd += cmd
+            # `kubectl exec` + subprocess w/ list of args has unexpected
+            # side-effects.
+            final_cmd = ' '.join(final_cmd)
```
Comment (richardliaw) [inline python/ray/autoscaler/command_runner.py:225]:
```suggestion
            final_cmd = " ".join(final_cmd)
```

File: python/ray/tests/test_command_runner.py [hunk 2]
```diff
@@ -125,10 +125,7 @@ def test_kubernetes_command_runner():
         """\'true && source ~/.bashrc && export OMP_NUM_THREADS=1 PYTHONWARNINGS=ignore && (export var1=\'"\'"\'"quote between this \\" and this"\'"\'"\';export var2=\'"\'"\'"123"\'"\'"\';echo helloo)\'"""  # noqa: E501
     ]
 
-    # Much easier to debug this loop than the function call.
-    for x, y in zip(process_runner.calls[0], expected):
-        assert x == y
-    process_runner.assert_has_call("1.2.3.4", exact=expected)
+    assert process_runner.calls[0] == ' '.join(expected)
```
Comment (richardliaw) [inline python/ray/tests/test_command_runner.py:128]:
```suggestion
    assert process_runner.calls[0] == " ".join(expected)
```

**Review by richardliaw — APPROVED** [review]
i have no way of testing this right now but i'm assuming you have

**Comment by richardliaw** [comment]
cc @twillkens

**Comment by wuisawesome** [comment]
Don't have a full k8's cluster, but I tested with minikube, but that should be enough for this.

**Comment by richardliaw** [comment]
sg, ping when tests pass

**Comment by richardliaw** [comment]
@wuisawesome can you list this on @barakmich release doc?

**Comment by wuisawesome** [comment]
> @wuisawesome can you list this on @barakmich release doc?

It's there :)

**Pull request merged**